File: client-app/src/app/models/problem.ts
export interface IProblem {
  Id: string;
  Title: string;
  AuthorEmail: string;
  Difficulty: string;
  GeneralDescription: string;
  IDescription: string;
  ODescription: string;
  TimeLimitInMilliseconds: Number;
  MemoryLimitInKiloBytes: Number;
  SampleInput: string;
  SampleOutput: string;
  Tags: string[];
  Hints: string[];
  Comments: IComment[];
  ProblemSetId: string;
}

export interface IComment {
  Id: string;
  CreatedAt: Date;
  Body: string;
  AuthorEmail: string;
  AuthorName: string;
}
export interface IProblemFormValues extends Partial<IProblem> {}

export class ProblemFormValues {
  Id: string = "";
  Title: string = "";
  Difficulty: string = "";
  GeneralDescription: string = "";
  IDescription: string = "";
  ODescription: string = "";
  Tags: string[] = [];
  Hints: string[] = [];
  SampleInput: string = "";
  ProblemSetId: string = "";
  SampleOutput: string = "";
  TimeLimitInMilliseconds: Number = 0;
  MemoryLimitInKiloBytes: Number = 0;
  AuthorEmail: string = "";
  TimeFactor: Number = 2;
  MemoryFactor: Number = 10;
  constructor(init?: Partial<ProblemFormValues>) {
    Object.assign(this, init);
  }
}

File: DataContracts/Problems/Problem.cs
using DataContracts.Comments;

namespace DataContracts.Problems
{
    public class Problem
    {
        public int Id { get; set; }

        public string Title { get; set; }

        public string AuthorEmail { get; set; }

        public string GeneralDescription { get; set; }

        public string InputDescription { get; set; }

        public string OutputDescription { get; set; }

        public string SampleInput { get; set; }

        public string SampleOutput { get; set; }

        public int TimeLimitInMilliseconds { get; set; }

        public int MemoryLimitInKiloBytes { get; set; }

        public string[] Tags { get; set; }

        public Difficulty Difficulty { get; set; }

        public string[] Hints { get; set; }

        public bool IsPublic { get; set; }
    }

    public enum Difficulty
    {
        Easy,
        Medium,
        Hard
    }
}